File: lista-1-python-go/exercicio-6/6-socket-client.go
// socket-client project 1-socket-client.go
package main

import (
	"fmt"
	"net"
)

const (
	SERVER_HOST = "localhost"
	SERVER_PORT = "12345"
	SERVER_TYPE = "tcp"
)

func server_send(connection net.Conn, err error, message string) {
	_, err = connection.Write([]byte(message))
	buffer := make([]byte, 1024)
	mLen, err := connection.Read(buffer)
	if err != nil {
		fmt.Println("Error reading:", err.Error())
	}
	fmt.Println("Received: ", string(buffer[:mLen]))
}

func server_receive(connection net.Conn, err error) {
	buffer := make([]byte, 1024)
	mLen, err := connection.Read(buffer)
	if err != nil {
		fmt.Println("Error reading:", err.Error())
	}
	fmt.Println("Received: ", string(buffer[:mLen]))
}

func main() {
	var employee_name string
	var employee_position string
	var employee_salary string
	var number_of_dependents string

	// establish connection
	connection, err := net.Dial(SERVER_TYPE, SERVER_HOST+":"+SERVER_PORT)

	if err != nil {
		panic(err)
	}

	fmt.Println("Enter the employee name: ")
	fmt.Scanln(&employee_name)
	fmt.Println("Enter the employee position: A, B, C or D")
	fmt.Scanln(&employee_position)
	fmt.Println("Enter the employee salary: $")
	fmt.Scanln(&employee_salary)
	fmt.Println("Enter the number of dependents: $")
	fmt.Scanln(&number_of_dependents)

	server_send(connection, err, employee_name)
	server_send(connection, err, employee_position)
	server_send(connection, err, employee_salary)
	server_send(connection, err, number_of_dependents)
	server_receive(connection, err)

	defer connection.Close()
}

File: lista-1-concorrente/exercicio-6/6-socket-client.py
import socket

SERVER = "127.0.0.1"
PORT = 12345

client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
client.connect((SERVER, PORT))

while True:
    name = input("Enter your name: ")
    client.send(name.encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    level = input("Enter your level (A, B, C, D): ")
    client.send(level.encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    salary = input("Enter your salary: ")
    client.send(salary.encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    number_of_dependents = input("Enter the number of dependents: ")
    client.send(number_of_dependents.encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    print("SERVER RESPONSE: " + client.recv(1024).decode())

    break
client.close()
